upf/upf_gtpu_encap.c

```diff
@@ -57,6 +57,272 @@ typedef enum
   UPF_ENCAP_N_NEXT,
 } upf_encap_next_t;
 
+static u16
+encode_ext_header_udp_port (vlib_buffer_t * b)
+{
+  u8 *p = vlib_buffer_get_current (b);
+  gtpu_ext_header_t *hdr = (gtpu_ext_header_t *)p;
+
+  hdr->type = GTPU_EXT_HEADER_UDP_PORT;
+  hdr->len = 1;   /* in 4 byte units */
+  *((u16 *)&hdr->pad) = clib_host_to_net_u16 (GTPU_UDP_PORT);
```

Comment:
This needs to be the original source port, not the fixed GTP-U port.

Sorry for giving the wring sample in the suggested change.

Reply:
Fixed